cornell-dti/idol-backend
# Add API response types to member endpoints and create member/team type file

### Summary 

This PR is the first step in refactoring the IDOL backend API that adds data types for members and teams in addition to creating response types for the member endpoints. 

In addition, specific HTTP status codes have also been added to the member endpoints:
- 200: Successful request
- 400: Empty email in request body
- 401: Current user does not exist in database
- 403: Current user does not have permission
- 404: Requested user doesn't exist 
- 500: Any error with database 

### Test Plan

Same test plan as PR #9:
- Check if the user can update their general info correctly
- Check if user can only update their information and not another user's (unless their role is admin or lead)
- Check if only admins/leads can update the role
- Check if only admins/leads can update the first name/last name  

### Notes
- Not sure if there is a way that the data and api types in the .d.ts files can be auto-imported into each file
- The data types for members, teams, and roles have only been moved and remain the same 
- The status field for response bodies have been changed to a number (from string) that is the HTTP status code
